### PR TITLE
feat: implement all `<dimension>` types

### PR DESCRIPTION
**Pull request by alexander-akait**
**Description:**

Refactor `<dimension>`, now we have all CSS types - https://www.w3.org/TR/css-values-4/#lengths and https://www.w3.org/TR/css-values-4/#other-units

We will use grammar parsing and for better minification - `1000ms` == `0.1s`, I will start to implement first transformations in `swc_css_minifier` when we merge it

**BREAKING CHANGE:**

Yes

**Related issue (if exists):**

No

### CODE REVIEW

File: crates/swc_css_parser/src/parser/value/mod.rs [hunk 1]
```diff
@@ -637,7 +637,43 @@ where
         let span = self.input.cur_span()?;
 
         if !is!(self, Dimension) {
-            return Err(Error::new(span, ErrorKind::Expected("Dimension")));
+            return Err(Error::new(span, ErrorKind::Expected("dimension token")));
+        }
+
+        match cur!(self) {
+            Token::Dimension { unit, .. } => {
+                match &*unit.to_ascii_lowercase() {
```
Comment (kdy1) [inline crates/swc_css_parser/src/parser/value/mod.rs:645]: This code seems bit problematic, as we don't use value of `&*unit.to_ascii_lowercase()`.

File: crates/swc_css_parser/src/parser/value/mod.rs [hunk 2]
```diff
@@ -637,7 +637,43 @@ where
...
+                    // <length>
+                    _ if is_length_unit(unit) => Ok(Dimension::Length(self.parse()?)),
```
Comment (kdy1) [inline crates/swc_css_parser/src/parser/value/mod.rs:647]: I think it should be`unit if is_length_unit(unit)`?

**Review by kdy1 — CHANGES_REQUESTED** [review]

**Comment by alexander-akait** [comment]
@kdy1 Done :+1:

**Review by kdy1 — APPROVED** [review]
Thanks!

**Pull request merged**